test/config/index.spec.js

```diff
@@ -5,36 +5,96 @@ describe('config/index', () => {
   describe('.parseConfigs(env, defaultArgv)', () => {
     let configParser;
     let defaultArgv;
+    let ghGot;
+    let glGot;
     beforeEach(() => {
       jest.resetModules();
       configParser = require('../../lib/config/index.js');
       defaultArgv = argv();
+      jest.mock('gh-got');
+      jest.mock('gl-got');
+      ghGot = require('gh-got');
+      glGot = require('gl-got');
     });
-    it('throws for no token', () => {
+    it('throws for no token', async () => {
       const env = {};
-      configParser.parseConfigs.bind(configParser, env, defaultArgv).should.throw('At least one repository must be configured');
+      let err;
+      try {
+        await configParser.parseConfigs(env, defaultArgv);
+      } catch (e) {
+        err = e;
+      }
+      expect(err.message).toBe('At least one repository must be configured');
     });
-    it('supports token in env', () => {
+    it('supports token in env', async () => {
       const env = { GITHUB_TOKEN: 'abc' };
-      configParser.parseConfigs.bind(configParser, env, defaultArgv).should.throw('At least one repository must be configured');
+      let err;
+      try {
+        await configParser.parseConfigs(env, defaultArgv);
+      } catch (e) {
+        err = e;
+      }
+      expect(err.message).toBe('At least one repository must be configured');
     });
-    it('supports token in CLI options', () => {
-      const env = {};
+    it('supports token in CLI options', async () => {
       defaultArgv = defaultArgv.concat(['--token=abc']);
-      configParser.parseConfigs.bind(configParser, env, defaultArgv).should.throw('At least one repository must be configured');
+      const env = { GITHUB_TOKEN: 'abc' };
+      let err;
+      try {
+        await configParser.parseConfigs(env, defaultArgv);
+      } catch (e) {
+        err = e;
+      }
+      expect(err.message).toBe('At least one repository must be configured');
+    });
+    it('autodiscovers github platform', async () => {
+      const env = { GITHUB_TOKEN: 'abc' };
+      defaultArgv = defaultArgv.concat(['--autodiscover', '--token=abc']);
+      ghGot.mockImplementationOnce(() => ({
+        body: [
+          {
+            full_name: 'a/b',
+          },
+          {
+            full_name: 'c/d',
+          },
+        ],
+      }));
+      await configParser.parseConfigs(env, defaultArgv);
+      expect(ghGot.mock.calls.length).toBe(1);
+      expect(glGot.mock.calls.length).toBe(0);
+    });
+    it('autodiscovers gitlab platform', async () => {
+      const env = { GITLAB_TOKEN: 'abc' };
+      defaultArgv = defaultArgv.concat(['--autodiscover', '--platform=gitlab', '--token=abc']);
```

Comment:
Fun fact, I tried a token with a leading dash, `-`, and it crashed `babel` transpiler.